benches/src/access_control.rs

```diff
@@ -50,7 +50,9 @@ pub async fn bench() -> eyre::Result<Report> {
         .wallet(EthereumWallet::from(bob.signer.clone()))
         .on_http(bob.url().parse()?);
 
-    let contract_addr = deploy(&alice).await;
+    let contract_addr = deploy(&alice).await?;
+    crate::cache_contract(&alice, contract_addr)?;
```

Comment:
Maybe this can be part of `deploy`? And maybe change `deploy`'s signature to receive a `enum CachePolicy { None, Bid(u64) }` which determines whether we cache or not? (this may seem overengineered, but it is forward-compatible)